### ISSUE TITLE
"Error: Get config Error: Command failed with exit code 13 (EACCES): /.../node_modules/prisma2/query-engine-darwin cli [...]"

### ISSUE THREAD

**Issue by janpio** [prisma/prisma2#668]
When running `prisma init foo` + `Blank Project | SQLite | Confirm | JavaScript | Demo Script` on CI (Azure Pipelines, Linux `Linux fv-az679 4.15.0-1059-azure #64-Ubuntu SMP Fri Sep 13 17:02:44 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux` and macOS `Darwin Mac-612.local 17.7.0 Darwin Kernel Version 17.7.0: Sun Jun  2 20:31:42 PDT 2019; root:xnu-4570.71.46~1/RELEASE_X86_64 x86_64`) it unfortunately crashes with `Error: Get config Error: Command failed with exit code 13 (EACCES): /Users/vsts/foo/node_modules/prisma2/query-engine-darwin cli [...]`:

```
2019-10-06T01:30:10.7505770Z bash-3.2$ prisma2 init foo
2019-10-06T01:30:25.9102030Z [?25l
2019-10-06T01:30:25.9104500Z Preparing your demo script [1m(JavaScript)[22m ...
2019-10-06T01:30:25.9104970Z 
2019-10-06T01:30:25.9106310Z [32mâœ”  Downloading the demo script from GitHub ...[39m
2019-10-06T01:30:25.9107260Z [32mâœ”  Extracting content to [1mfoo[22m ...[39m
2019-10-06T01:30:25.9108120Z â ¸  Installing dependencies with: `npm install` ...
2019-10-06T01:30:25.9108840Z [2m4  Preparing your database ...[22m
2019-10-06T01:30:25.9109730Z [1m[37m[41m ERROR [49m[39m[22m [31mError during command execution[39m
2019-10-06T01:30:25.9110240Z 
2019-10-06T01:30:25.9111610Z Error: Get config Error: Command failed with exit code 13 (EACCES): /Users/vsts/foo/node_modules/prisma2/query-engine-darwin cli --get_config {"datamodel":"generator photon {\n  provider = \"photonjs\"\n}\n\ndatasource db {\n  provider = \"sqlite\"\n  url      = \"file:dev.db\"\n}\n\nmodel User {\n  id    String  @default(cuid()) @id\n  email String  @unique\n  name  String?\n  posts Post[]\n}\n\nmodel Post {\n  id        String   @default(cuid()) @id\n  createdAt DateTime @default(now())\n  updatedAt DateTime @updatedAt\n  published Boolean  @default(true)\n  title     String\n  content   String?\n  author    User?\n}"}
2019-10-06T01:30:25.9112150Z 
2019-10-06T01:30:25.9112980Z spawn /Users/vsts/foo/node_modules/prisma2/query-engine-darwin EACCES
2019-10-06T01:30:25.9113370Z error Command failed with exit code 1.
```
![image](https://user-images.githubusercontent.com/183673/66262964-ee6d1300-e7ea-11e9-9daf-1b74bfe83673.png)

---

I think this could be related to the file not being executable, these are the file permissions in that folder:

```
# macOS

ls -l /Users/vsts/foo/node_modules/prisma2/
-rw-r--r--   1 vsts  staff   759 Oct  6 01:50 README.md
drwxr-xr-x  19 vsts  staff   646 Oct  6 01:50 build
drwxr-xr-x   3 vsts  staff   102 Oct  6 01:50 download-build
-rw-r--r--   1 vsts  staff   103 Oct  6 01:50 migration-engine
drwxr-xr-x   8 vsts  staff   272 Oct  6 01:50 nexus_prisma_ncc_build
-rw-r--r--   1 vsts  staff  2447 Oct  6 01:50 package.json
-rw-r--r--   1 vsts  staff    99 Oct  6 01:50 query-engine-darwin
drwxr-xr-x  16 vsts  staff   544 Oct  6 01:50 runtime

# Linux

ls -l   /home/vsts/foo/node_modules/prisma2/
drwxr-xr-x 15 vsts docker 4096 Oct  6 01:57 build
drwxr-xr-x  2 vsts docker 4096 Oct  6 01:57 download-build
-rw-r--r--  1 vsts docker  103 Oct  6 01:57 migration-engine
drwxr-xr-x  5 vsts docker 4096 Oct  6 01:57 nexus_prisma_ncc_build
-rw-r--r--  1 vsts docker 2447 Oct  6 01:57 package.json
-rw-r--r--  1 vsts docker   99 Oct  6 01:57 query-engine-linux-glibc-libssl1.1.0
-rw-r--r--  1 vsts docker  759 Oct  6 01:57 README.md
drwxr-xr-x  3 vsts docker 4096 Oct  6 01:57 runtime
```

If I read that correctly, the downloaded engines are not executable but should be?

---

I guess the problem is triggered while `"postinstall": "prisma2 generate"` is running, but might be caused earlier during installation.

As `prisma2 init` doesn't show the `npm install` output, I don't know exactly what the last output there was. Can I somehow get more information there to understand where the permissions go wrong?

PS: On Windows it works just fine :D

**Comment by pantharshit00** [comment]
Can you please try using `--unsafe-perm` when installing prisma2 on that instance?

**Comment by emolr** [comment]
Using `prisma2@2.0.0-preview-13.2` I can run `prisma2 generate` succesfully from my terminal, but it fails like above when being run as a `postinstall` script. 

In my package.json I also have:
```
"config": {
    "unsafe-perm": true
  }
```

Doesn't seem to make a difference though.

Note! 
My prisma2 is install globally with `sudo`

**Comment by janpio** [comment]
@emolr What OS and version are you using? How exactly did you install the CLI? How did you create your project exactly? I would really like to be able to reproduce this in a simpler envirenment.

**Comment by emolr** [comment]
@janpio 

OS: Im using MacOS Catalina version 10.15
Global Installation process: `sudo npm i -g --unsafe-perm prisma2@2.0.0-preview-13.2`.
Project created using `prisma2 init`on version `2.0.0-preview-12`

But the issue here is with the local version of prisma2 installed via `npm install prisma2@2.0.0-preview-13.2` and run by `scripts: { generate: 'prisma2 generate' }`.

Are there other informations i can give you to help you reproduce the issue?

**Comment by pantharshit00** [comment]
Maybe also post a screenshot as well :)

**Comment by pantharshit00** [comment]
I think it is more related to when you storing your npm cache and local registry.

If you use `sudo` with npm on a mac, it would store it in context of root. Then you would need `--unsafe-perm` as npm won't allow any script to run in context with root without explicit confirmation.

**Comment by lvauvillier** [comment]
Same issue.
Fresh a blank project  on Catalina
I installed prisma2 globally using `sudo npm install -g prisma2 --unsafe-perm`.

I tryed also to run directly `prisma2 generate` and it fails with the same error

**Comment by lvauvillier** [comment]
As a workaround, i setup npm with a prefix to install global packages without the need of `sudo` in a user related directory. (http://michaelb.org/the-right-way-to-do-global-npm-install-without-sudo/)

**Comment by joshuarobs** [comment]
@lvauvillier I tried to follow the instructions in that link and still it didn't work for me. Was there anything else that you did to make it work?

**Comment by lvauvillier** [comment]
@joshuarobs  I update to prisma2@2.0.0-preview014.2 and the trick didn't work anymore. 
My new workaround is to use a docker container.

**Comment by BjoernRave** [comment]
Is it maybe possible that there is a connection between this error and this npm config: `npm config set ignore-scripts true`  ?

**Comment by pantharshit00** [comment]
Yes, it might be possible @BjoernRave. In the case, you will need to run our download script manually so that you can download the prisma2 binaries.

**Comment by janpio** [comment]
Wow, good catch @BjoernRave. Can you open a new issue about this please? Then we will formally reproduce this and then decide on what we can do about it. We should probably at least print a warning with instructions on how to do it manually or via a command of the CLI.

**Comment by lvauvillier** [comment]
@divyenduz I still had this issue in preview20, but spend some time to find what is going on:
`prisma2 generates` fails was due to an absence of the engine binaries.

The problem was during installation: npm my cache folder was not writable (wrong permissions) and the `download-build.js` process fails silently whereas the cli indicates 100% downloaded and the flag file `download-done` created. 

Maybe add an explicit message/error here will help.

**Comment by pantharshit00** [comment]
Can you please open a new issue about this @lvauvillier? I remember we try to download the binary on the first run if we fail silently but we need to validate.
